tests/Integration/Behaviour/Features/Context/Domain/Product/SpecificPricePrioritiesFeatureContext.php

```diff
@@ -52,59 +54,106 @@ public static function restoreSpecificPricesPrioritiesAfterFeature(): void
+     * @When I set following default specific price priorities:
+     *
+     * @param PriorityList $priorityList
      */
-    public function setGlobalPriorities(TableNode $prioritiesTable): void
+    public function setDefaultPriorities(PriorityList $priorityList): void
     {
-        $priorities = $prioritiesTable->getRow(0);
+        /** @var SpecificPricePriorityUpdater $priorityUpdater */
+        $priorityUpdater = CommonFeatureContext::getContainer()
+            ->get('prestashop.adapter.product.specific_price.update.specific_price_priority_updater')
+        ;
 
         try {
-            $this->getCommandBus()->handle(new SetGlobalSpecificPricePriorityCommand($priorities));
+            $priorityUpdater->updateDefaultPriorities($priorityList);
         } catch (DomainException $e) {
             $this->setLastException($e);
         }
     }
 
     /**
+     * @see transformPriorityList
+     *
+     * @Given default specific price priorities are set to following:
+     * @Then default specific price priorities should be the following:
+     *
+     * @param PriorityList $priorityList
+     */
+    public function assertDefaultPriorities(PriorityList $priorityList): void
+    {
+        $this->assertPriorities($priorityList, null);
+    }
+
+    /**
+     * @see transformPriorityList
+     *
      * @Then product :productReference should have following specific price priorities:
      *
      * @param string $productReference
-     * @param TableNode $prioritiesTable
+     * @param PriorityList $priorityList
+     */
+    public function assertProductPriorities(string $productReference, PriorityList $priorityList): void
+    {
+        $this->assertPriorities($priorityList, $productReference);
+    }
+
+    /**
+     * @Transform table:priorities
+     *
+     * @param TableNode $tableNode
+     *
+     * @return PriorityList
+     */
+    public function transformPriorityList(TableNode $tableNode): PriorityList
+    {
+        $prioritiesTable = $tableNode->getColumnsHash();
+        $priorities = [];
+        foreach ($prioritiesTable as $value) {
+            $priorities[] = $value['priorities'];
+        }
+
+        return new PriorityList($priorities);
+    }
+
+    /**
+     * @param PriorityList $expectedPriorityList
+     * @param string|null $productReference when null it checks default priorities list
      */
-    public function assertProductPriorities(string $productReference, TableNode $prioritiesTable): void
+    private function assertPriorities(PriorityList $expectedPriorityList, ?string $productReference)
     {
-        $expectedPriorities = $prioritiesTable->getRow(0);
         SpecificPrice::flushCache();
-        $actualPriorities = SpecificPrice::getPriority($this->getSharedStorage()->get($productReference));
+        $productId = $productReference ? $this->getSharedStorage()->get($productReference) : 0;
+        $actualPriorities = SpecificPrice::getPriority($productId);
```

Comment:
The problem with this way of checking is that `SpecificPrice::getPriority` always returns something even for product which have no custom priorities It might be a good idea to keep this one but we also need a check related to `GetProductForEditing` which checks that the priorities returned is null or empty

Or else we don't test that part of the query handler and we can't be sure that the product custom values are correctly removed when we use the remove command (we can't tell the difference between a custom configuration matching the default one and an empty one this way)